### PR TITLE
fix(iteration): Clear iterationsValue before loading new values

### PR DESCRIPTION
**Pull request by jarifibrahim**
### Mandatory
- [x] What issue/task does this PR references?
This PR fixes  https://github.com/openshiftio/openshift.io/issues/3318
- [x] Are the tests Included?
Yes

Thanks for the help @sahil143

### CODE REVIEW

**Comment by jarifibrahim** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

File: src/tests/ui/planner/iteration.ts [hunk 1]
```diff
@@ -22,11 +23,15 @@ export class Iteration extends ui.BaseElement {
...
     
-  async addNewIteration(iterationName: string, parentIteration: string ) {
+  async addNewIteration(iterationName: string, parentIteration?: string, withDates?: boolean) {
     await this.iterationName.enterText(iterationName);
-    await this.parentIteration.enterText(parentIteration);
-    await this.parentIteration.clickWhenReady();
-    await this.selectCalendarDate();    
+    if(parentIteration) {
+      await this.parentIteration.enterText(parentIteration);
+      await this.parentIteration.clickWhenReady();
```
Comment (Raunak1203) [inline src/tests/ui/planner/iteration.ts]: @jarifibrahim change await this.parentIteration.clickWhenReady(); to await this.parentIteration.select(parentIteration)

Comment (jarifibrahim) [reply src/tests/ui/planner/iteration.ts]: Done

**Comment by jarifibrahim** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

**Comment by nimishamukherjee** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

**Pull request merged**